### DISCUSSION TITLE
How do I fix this because it is taking too much time "Deploying gh-pages branch to website (this may take a few minutes)"?

### DISCUSSION THREAD

**Discussion by muhaksim** [quarto-dev/quarto-cli#7460] [Q&A] [answered]
### Description

I wanted to publish my site on github pages and it did publish but the command is still loading in my terminal saying:
_"Deploying gh-pages branch to website (this may take a few minutes)"_.

Any solution for this?

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

Unfortunately, without any information on your environment or document, there is absolutely nothing we can do.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by muhaksim** [comment]
I am building a blog with Quarto. What pieces should I share or should I share the whole project?

**Reply by mcanouil** [reply]
The minimum amount for us to reproduce.

**Comment by cderv** [comment]
In case of publishing, I understand this is hard to share an exact example. This step you are blocked in is the last one waiting for the website to be available and loaded so that we can open it in the browser safely. 

This means that your website should already be push into the `gh-pages` branch of your repo. Can you check that ? 

Next step is that Github should have deployed it to its `.io` page. Can you check the Action page of your repo to see if there is an error ? 

If it is not deployed, then one thing is to check the repo configuration to see if you correctly followed the recommandation
https://quarto.org/docs/publishing/github-pages.html#publish-command

Github pages should be deployed from Source and from the `gh-pages` branch. 

If you can share your repo, we can try having a look and see if we see anything unusual.

Hope it helps.

**Comment by muhaksim** [comment]
I think it is pushed to `gh-pages` branch. My problem was it was it kept loading saying `Deploying gh-pages branch to website (this may take a few minutes)`.

Here is the repo:
https://github.com/muhaksim/muhaksim.github.io

**Reply by mcanouil** [reply]
Was it a one time thing?
Can you reproduce it yourself by trying to publish again?

**Reply by JosiahParry** [reply]
I have had the same problem. I ran it to generate the _publish.yml file. It ran for 22 minutes before I cancelled it. It published to gh-pages but never produced the _publish.yml file as it says here: https://quarto.org/docs/publishing/github-pages.html#publish-action

I ran this because the publish github action that I borrowed from [quarto-actions](https://github.com/quarto-dev/quarto-actions/blob/main/examples/quarto-publish-example.yml) failed to execute. 

I would suspect that the reprex is to create a new github repo, create a new quarto website, initialize the git repo, do an initial push. Then run the command.

**Reply by JosiahParry** [reply]
I did just update quarto to 1.3.450 yesterday. I had never used this command before though. Unsure if its related.

**Reply by JosiahParry** [reply]
Note that after running it a second time it executed to completion. However, again, no `_publish.yml` file was generated in the original directory or the `gh-pages` branch it created.

**Reply by mcanouil** [reply]
@JosiahParry See (especially the ones labelled https://github.com/quarto-dev/quarto-cli/labels/documentation):
- https://github.com/quarto-dev/quarto-cli/issues?q=is%3Aissue+is%3Aopen+gh-pages+label%3Apublishing-github

**Reply by JosiahParry** [reply]
I'm not sure what I'm supposed to see 🤷🏼

**Reply by mcanouil** [reply]
TL;DR: The documentation is currently outdated.

Also, this is unrelated to the initial question.

**Comment by muhaksim** [comment, marked as answer]
I fixed the problem. I deleted the blog locally and the repo from GitHub as well. I re-initialized it again and used `quarto publish gh-pages`.

Thanks, @mcanouil, @JosiahParry and @cderv for your time. I really appreciate it.

**Comment by pstorozenko** [comment]
I experienced the same issue with quarto 1.5.57, I really didn't want to remove the repo, and I've found the solution!

In my case, I had to specify the `gh-pages` branch manually in the repo settings `https://github.com/<org>/<repo>/settings/pages`

![image](https://github.com/user-attachments/assets/5d0ee430-f76f-4b51-b4bc-0e1224ac3656)

Maybe it helps someone, cheers!

**Reply by cderv** [reply]
THanks for sharing ! 

For more context, we document the setup to do at: https://quarto.org/docs/publishing/github-pages.html#source-branch

This include checking this `gh-pages` config.

**Reply by pstorozenko** [reply]
Good point, I missed this part of docs 🤦🏻
